src/generator/generator_visitor.ts

```diff
@@ -391,6 +403,35 @@ class GeneratorVisitor extends Visitor {
         return this.module.i32.store(0, 4, this.getPointer(), value);
     }
 
+    private createLocalArray(val: LVal, length: number) {
+        if (!isIdentifier(val)) {
+            throw new Error('Assigned to unknown variable');
+        }
+
+        const memoryLocation = this.getArrayMemoryLocation(val.name);
+        const lengthAddress = this.module.i32.const(memoryLocation - 4);
+        // @ts-ignore because store() returns an expression
+        this.statements.push(this.module.i32.store(0, 4, lengthAddress, this.module.i32.const(length)));
+
+        for (let i: number = length - 1; i >= 0; i--) {
```

Comment:
Is it possible that the reverse order has a negative impact on the performance?

Reply:
And you should remove the type of `i`.

Reply:
I changed it to a normal for loop and now use the length to calculate the offset.

Reply:
This doesn't solve the problem. The stores are still in descending order. I think we should leave it for now. But I would revert the change with the offset calculation.

Reply:
The problem is, that we will have at least one reverse for loop. Either in the `createLocalArray` or the `visitArrayExpression` method. If we push them in normal order, they need to be poped in reverse order. If we push them in reverse order, they can be poped in normal order.